DOC: updated docstring for nanoseconds function per doc guidelines

chalmerlowe opened the pull request:
- [ ] closes #xxxx
- [ ] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

```
################################################################################
################### Docstring (pandas.Timedelta.nanoseconds) ###################
################################################################################

Return the number of nanoseconds (n), where 0 <= n < 1 microsecond.

Returns
-------
int : Number of nanoseconds

See Also
--------
Timedelta.components : Return all attributes with assigned values (i.e.
    days, seconds, microseconds, nanoseconds)

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	No extended summary found
	No examples section found
```

chalmerlowe committed:
updated docstring for nanoseconds function

chalmerlowe committed:
Merge remote-tracking branch 'upstream/master' into nanosecond-docstring

chalmerlowe committed:
DOC: updated docstring for nanoseconds function per doc guidelines

WillAyd reviewed (changes requested):
Thanks for the changes - an Examples section would be a nice touch

chalmerlowe committed:
DOC: added examples to nanoseconds function and minor formatting edits

chalmerlowe committed:
DOC: added example header

chalmerlowe committed:
DOC: added example header and space

chalmerlowe committed:
DOC: fixed minor spelling error

WillAyd reviewed (approved):
lgtm - thanks!

chalmerlowe committed:
removed trailing whitespace, long lines

chalmerlowe committed:
Merge remote-tracking branch 'upstream/master' into nanosecond-docstring

chalmerlowe committed:
DOC: updates to  docstring for nanoseconds function per doc guidelines

chalmerlowe commented:
* Removed the trailing whitespace.
* Modified linebreaks to ensure that the text was within line length limits.
* Ran the `git diff upstream/master -u -- "*.py" | flake8 --diff` command as required in the documentation guide.


```
################################################################################
################### Docstring (pandas.Timedelta.nanoseconds) ###################
################################################################################

Return the number of nanoseconds (n), where 0 <= n < 1 microsecond.

Returns
-------
int
    Number of nanoseconds.

See Also
--------
Timedelta.components : Return all attributes with assigned values (i.e.
    days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds).

Examples
--------
**Using string input**

>>> td = pd.Timedelta('1 days 2 min 3 us 42 ns')
>>> td.nanoseconds
42

**Using integer input**

>>> td = pd.Timedelta(42, unit='ns')
>>> td.nanoseconds
42

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	No extended summary found
```

jreback commented on pandas/_libs/tslibs/timedeltas.pyx:
```diff
@@ -791,9 +791,32 @@ cdef class _Timedelta(timedelta):
+       
+        Returns
+        -------
+        int
```
shouldn't this be on the same line with a colon? @TomAugspurger @jorisvandenbossche

TomAugspurger replied:
This is fine as is. It's typically

```
name : type
    Explanation
```

but when the name is obvious (it'd just be `nanoseconds` here), it can be omitted.

chalmerlowe commented:
Thanks for closing. My **First PR** on pandas.

There was some back and forth comments on the formatting...

I followed the examples in the [docstrings guide: ](https://python-sprints.github.io/pandas/guide/pandas_docstring.html)

**With a single return value**
```
Returns
-------
float
    Random number generated.
```

**With multiple return values**
```
Returns
-------
length : int
    Length of the returned string.
letters : str
    String of random letters.
```
